src/pip/_internal/exceptions.py

```diff
@@ -3,9 +3,12 @@
 import configparser
 import re
 from itertools import chain, groupby, repeat
-from typing import TYPE_CHECKING, Dict, Iterator, List, Optional
+from typing import TYPE_CHECKING, Dict, List, Optional, Union
```

Comment:
Now Python 3.7 is the minimum, instead of:

```python
from typing import Union
a: Union[int, str]
```

You can use the more concise (PEP 604):

```python
from __future__ import annotations
a: int | str
```

Similarly, you can use `dict` and `list` instead of `typing.Dict` and `typing.List` etc:

https://www.python.org/dev/peps/pep-0585/#implementation

Reply:
Let's do this in a blanket move, in a separate PR.

Reply:
#10725